plugin/evm/limit_order.go

```diff
@@ -135,67 +139,67 @@ func (lop *limitOrderProcesser) listenAndStoreLimitOrderTransactions() {
 	logsCh := make(chan []*types.Log)
 	logsSubscription := lop.backend.SubscribeHubbleLogsEvent(logsCh)
 	lop.shutdownWg.Add(1)
-	go lop.ctx.Log.RecoverAndPanic(func() {
+	go func() {
 		defer lop.shutdownWg.Done()
 		defer logsSubscription.Unsubscribe()
-
 		for {
 			select {
 			case logs := <-logsCh:
-				lop.mu.Lock()
-
-				lop.contractEventProcessor.ProcessEvents(logs)
-
-				lop.mu.Unlock()
+				recoverPanic(func() {
+					lop.mu.Lock()
+					defer lop.mu.Unlock()
```

Comment:
defer statement does't work in case block. that's why `Unlock` was written after `contractEventProcessor.ProcessEvents(logs)`